src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -144,62 +165,86 @@ private boolean isActiveSession(final String subscriptionId, final String stream
-        if (Cursor.BEFORE_OLDEST_OFFSET.equals(cursor.getOffset())) {
-            cursorToProcess = new SubscriptionCursor(cursor.getPartition(), "-1", cursor.getEventType(),
-                    cursor.getCursorToken());
+        } catch (final NakadiRuntimeException e) {
+            throw (InvalidCursorException) e.getException();
         }
-        final EventType eventType = eventTypeRepository.findByName(cursorToProcess.getEventType());
-        final NakadiCursor toProcess = new NakadiCursor(
-                eventType.getTopic(),
-                cursorToProcess.getPartition(),
-                cursorToProcess.getOffset());
-
-        topicRepository.validateCommitCursor(toProcess);
-        return commitCursor(subscriptionId, eventType.getTopic(), cursorToProcess);
     }
 
-    private boolean commitCursor(final String subscriptionId, final String eventType, final SubscriptionCursor cursor)
-            throws ServiceUnavailableException, NoSuchSubscriptionException, InvalidCursorException {
+    private List<Boolean> commitPartitionCursors(final String subscriptionId,
+                                                 final String topic,
+                                                 final List<NakadiCursor> cursors,
+                                                 final EventTypePartition etPartition)
+            throws ServiceUnavailableException {
 
-        final String offsetPath = format(PATH_ZK_OFFSET, subscriptionId, eventType, cursor.getPartition());
+        final String offsetPath = format(PATH_ZK_OFFSET, subscriptionId, topic, etPartition.getPartition());
         try {
             @SuppressWarnings("unchecked")
-            final Boolean committed = executeWithRetry(() -> {
+            final List<Boolean> committed = executeWithRetry(() -> {
                         final Stat stat = new Stat();
                         final byte[] currentOffsetData = zkHolder.get()
                                 .getData()
                                 .storingStatIn(stat)
                                 .forPath(offsetPath);
                         final String currentOffset = new String(currentOffsetData, Charsets.UTF_8);
-
-                        // Yep, here we are trying to hack a little. This code
-                        // should be removed during timelines implementation
-                        final NakadiCursor cursorToCommit = new NakadiCursor(eventType, cursor.getPartition(),
-                                cursor.getOffset());
-                        final NakadiCursor currentCursor = new NakadiCursor(eventType, cursor.getPartition(),
+                        NakadiCursor currentMaxCursor = new NakadiCursor(topic, etPartition.getPartition(),
                                 currentOffset);
-                        if (topicRepository.compareOffsets(cursorToCommit, currentCursor) > 0) {
+
+                        final List<Boolean> commits = Lists.newArrayList();
```

Comment:
I don't think we should have that in NakadiCursor. In most cases where NakadiCursor is used it's not about committing, so it will always have some additional field that is not used.